stl/inc/generator

```diff
@@ -311,14 +311,13 @@ private:
             _STL_INTERNAL_STATIC_ASSERT(is_pointer_interconvertible_base_of_v<_Base, _CoroPromise>);
 #endif // ^^^ no workaround ^^^
 
-            _Base& _Current = _Handle.promise();
-            if (!_Current._Info) {
+            _Nest_info* _Info = _Handle.promise()._Try_get_nest_info();
+            if (!_Info) {
                 return _STD noop_coroutine();
             }
 
-            coroutine_handle<_Base> _Cont        = _Current._Info->_Parent;
-            _Current._Info->_Root.promise()._Top = _Cont;
-            _Current._Info                       = nullptr;

```

Comment:
I see that `_Current._Info = nullptr;` (i.e. `_Handle.promise()._Info = nullptr;`) is being dropped, but I don't completely understand why.

The new `_Info->_Root.promise()._Set_top(_Cont);` (i.e. `_Handle.promise()._Try_get_nest_info()->_Root.promise()._Set_top(_Cont);`) seems to be modifying a different promise - are they actually the same?

(Note that I am very uninformed about the data structure relationships here.)

Reply:
> I see that `_Current._Info = nullptr;` (i.e. `_Handle.promise()._Info = nullptr;`) is being dropped, but I don't completely understand why.

We are in final awaiter, so current coroutine is about to be destroyed. Setting `_Current._Info` to `nullptr` is unnecessary.

> The new `_Info->_Root.promise()._Set_top(_Cont);` (i.e. `_Handle.promise()._Try_get_nest_info()->_Root.promise()._Set_top(_Cont);`) seems to be modifying a different promise - are they actually the same?

`_Info->_Root.promise()._Set_top(_Cont);` (used to be `_Current._Info->_Root.promise()._Top = _Cont;`) modifies root promise which in this situation cannot be the same as `_Handle.promise()`.

If `_Handle.promise()` was the same as root, then `_Handle.promise._Try_get_nest_info()` would be equal to `nullptr` (there is no nested generator), and in such case we return `noop_coroutine`.

Reply:
Got it, thanks! :heart_eyes_cat: